File: csharp/space-age/SpaceAge.cs
public record SpaceAge(ulong Seconds)
{
    private const ulong SecondsOnEarth = 31557600;

    public double OnEarth() => FromPeriodInEarthYears(1.0d);

    public double OnMercury() => FromPeriodInEarthYears(0.2408467d);

    public double OnVenus() => FromPeriodInEarthYears(0.61519726d);

    public double OnMars() => FromPeriodInEarthYears(1.8808158d);

    public double OnJupiter() => FromPeriodInEarthYears(11.862615d);

    public double OnSaturn() => FromPeriodInEarthYears(29.447498d);

    public double OnUranus() => FromPeriodInEarthYears(84.016846d);

    public double OnNeptune() => FromPeriodInEarthYears(164.79132d);

    private double FromPeriodInEarthYears(double period) => Seconds / period / SecondsOnEarth;
}

File: kotlin/space-age/src/main/kotlin/SpaceAge.kt
import kotlin.math.roundToInt

class SpaceAge(private val seconds: Long) {
    fun onMercury() = fromPeriodInEarthYears(0.2408467)
    fun onVenus() = fromPeriodInEarthYears(0.61519726)
    fun onEarth() = fromPeriodInEarthYears(1.0)
    fun onMars() = fromPeriodInEarthYears(1.8808158)
    fun onJupiter() = fromPeriodInEarthYears(11.862615)
    fun onSaturn() = fromPeriodInEarthYears(29.447498)
    fun onUranus() = fromPeriodInEarthYears(84.016846)
    fun onNeptune() = fromPeriodInEarthYears(164.79132)

    private fun fromPeriodInEarthYears(period: Double) = seconds.div(period).div(SECONDS_ON_EARTH).round()
    private fun Double.round() = times(100.0).roundToInt().div(100.0)
}

private const val SECONDS_ON_EARTH = 31_557_600L
